src/sources/syslog_remap.rs

```diff
@@ -0,0 +1,97 @@
+use crate::sources::socket::SocketConfig;
+use crate::{
+    config::{
+        log_schema, DataType, GenerateConfig, Resource, SourceConfig, SourceContext,
+        SourceDescription,
+    },
+    transforms::remap::{Remap, RemapConfig},
+    Pipeline,
+};
+use bytes::Bytes;
+
+use futures::StreamExt;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct SyslogRemapConfig {
+    // Config settings we may need access to
+    // host_key: Option<String>,
+    #[serde(flatten)]
+    original_config: SocketConfig,
+}
+
+inventory::submit! {
+    SourceDescription::new::<SyslogRemapConfig>("syslog_remap")
+}
+
+impl GenerateConfig for SyslogRemapConfig {
+    fn generate_config() -> toml::Value {
+        return SocketConfig::generate_config();
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "syslog_remap")]
+impl SourceConfig for SyslogRemapConfig {
+    async fn build(&self, mut cx: SourceContext) -> crate::Result<super::Source> {
+        let conf = RemapConfig {
+            source: r#"
+structured = parse_syslog!(.message)
+. = merge(., structured)
+"#
```

Comment:
We've started using the `indoc!` macro for things like this so they don't have to break the indentation flow like this.